presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -115,6 +116,34 @@ public void testCastFromDouble()
     }
 
     @Test
+    public void testCastToReal()
+            throws Exception
+    {
+        assertFunction("cast(JSON 'null' as REAL)", REAL, null);
+        assertFunction("cast(JSON '-128' as REAL)", REAL, -128.0f);
+        assertFunction("cast(JSON '128' as REAL)", REAL, 128.0f);
+        assertFunction("cast(JSON '1.234567890' as REAL)", REAL, 1.234567890f);
```

Comment:
We mean this:
```
assertFunction("cast(JSON '12345678901234567890' as REAL)", REAL, 1.2345679e19f);
```

The goal to have this huge number is to make sure it doesn't first cast into an BIGINT (which will cause Overflow). 

I will change it for the merge :)